starboard/android/shared/media_decoder.cc

```diff
@@ -317,18 +317,18 @@ void MediaDecoder::DecoderThreadFunc() {
                                   &dequeue_output_results);
       }
 
-      if (!tunnel_mode_enabled_) {
```

Comment:
This is when tunnel model is disabled.

Reply:
My understanding is code from line 320-334 are only used by non-tunnel mode. I still don't understand how this change can reduce cpu usage of tunnel mode. Could you provide us more context how this cl improves the cpu usage?

Reply:
@Rongo-JL Would you be able to help answer this? Thanks!

Reply:
Yes, if we turn tunnel mode on and play a video, dequeue_output_result is not empty.
Because format change is also queued in the dequeue_output_result, it needs to call host_->RefreshOutputFormat(media_codec_bridge_.get()); to consume it.
Or it will not go into condition_variable_.WaitTimed(kSbTimeMillisecond);
Therefore, this thread will be a busy loop which consumes CPU usage dramatically.

Reply:
Thanks for clarification.